x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,753 @@
+        this.syncNow = syncNow;
+    }
+
+    public Connector(StreamInput in) throws IOException {
+        this.connectorId = in.readString();
+        this.apiKeyId = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.configuration = in.readMap(StreamInput::readString, StreamInput::readGenericValue);
+        } else {
+            this.configuration = null;
+        }
+        if (in.readBoolean()) {
+            this.customScheduling = new ConnectorCustomSchedule(in);
+        } else {
+            this.customScheduling = null;
+        }
+        this.description = in.readOptionalString();
+        this.error = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.features = new ConnectorFeatures(in);
+        } else {
+            this.features = null;
+        }
+        this.indexName = in.readOptionalString();
+        this.isNative = in.readBoolean();
+        this.language = in.readOptionalString();
+        this.lastAccessControlSyncError = in.readOptionalString();
+        this.lastAccessControlSyncScheduledAt = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.lastAccessControlSyncStatus = ConnectorSyncStatus.valueOf(in.readString());
+        } else {
+            this.lastAccessControlSyncStatus = null;
+        }
+        this.lastDeletedDocumentCount = in.readOptionalLong();
+        this.lastIncrementalSyncScheduledAt = in.readOptionalString();
+        this.lastIndexedDocumentCount = in.readOptionalLong();
+        this.lastSeen = in.readOptionalString();
+        this.lastSyncError = in.readOptionalString();
+        this.lastSyncScheduledAt = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.lastSyncStatus = ConnectorSyncStatus.valueOf(in.readString());
+        } else {
+            this.lastSyncStatus = null;
+        }
+        this.lastSynced = in.readOptionalString();
+        this.name = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.pipeline = new ConnectorIngestPipeline(in);
+        } else {
+            this.pipeline = null;
+        }
+        if (in.readBoolean()) {
+            this.scheduling = new ConnectorScheduling(in);
+        } else {
+            this.scheduling = null;
+        }
+        this.serviceType = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.status = ConnectorStatus.valueOf(in.readString());
+        } else {
+            this.status = null;
+        }
+        if (in.readBoolean()) {
+            this.syncCursor = in.readGenericValue();
+        } else {
+            this.syncCursor = null;
+        }
+        this.syncNow = in.readBoolean();
+    }
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<Connector, String> PARSER = new ConstructingObjectParser<>(
+        "connector",
+        false,
+        (args, connectorId) -> new Builder().setConnectorId(connectorId)
```

Comment:
As mentioned, it might make sense to break this class into additional subclasses for easier parsing and attribute management.

Reply:
Agreed that looks ugly ...